backend/translation/views.py

```diff
@@ -182,18 +215,36 @@ def post(self, request):
                 status=status.HTTP_404_NOT_FOUND,
             )
 
-        # Return the appropriate response depending on whether a new translation was created or not
-        if created:
+        # If the translation belongs to the current user, update the translation
+        if translation.user == user:
+            translation.captions = captions
+            translation.translation_type=(
+                HUMAN_EDITED
+                if current_translation_type == MACHINE_GENERATED
+                else MANUALLY_CREATED
+            )
+            translation.save()
+            
```

Comment:
Also, for this case, is this the correct way to update the `translation_type`?

Reply:
Why do we need to update the `translation_type` ?
If the translation.user matches the request.user, it already means that it is not of machine_generated type (since translation.user will be None for "mg")

Reply:
The update was already there in Nishant's code where the type was being updated to `he` but @pranav2305 wanted it changed this way. Can you add the explanation here @pranav2305 ?

Reply:
> The update was already there in Nishant's code 

Ohh OK, I guess we can remove it.

Reply:
Removed